### ISSUE TITLE
SSO everything forbidden after login

### ISSUE THREAD

**Issue by DrissiReda** [tektoncd/dashboard#2214]
I'm deploying tekton dashboard and oauth2-proxy in kubernetes. I'm using keycloak that is inside my kubernetes cluster for sso. So the user goes through oauth2-proxy and is then redirected to tekton-dashboard after successful login. My configuration works and I can login successfully, but once I'm logged in the dashboard displays "Unauthorized" everywhere, and the logs of oauth2-proxy show this:

```
10.130.3.32 - b5ac70f2cd643462423b0df104264782 - secadmin@localhost [2021/09/29 10:18:35] tekton.localhost GET / 
"/apis/tekton.dev/v1beta1/tasks/" HTTP/1.1 "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:92.0) 
Gecko/20100101 Firefox/92.0" 401 165 0.002
```
I don't understand why, because I put the `tekton-dashboard` serviceAccount in my oauth2-proxy deployment.

Any ideas?

**Comment by AlanGreene** [comment]
This is most likely caused by an invalid bearer token being passed in the Authorization header on requests to the Dashboard upstream.

If you're not expecting the Dashboard to authenticate with the user's token instead of the Dashboard's own ServiceAccount you can prevent the proxy from forwarding the header on upstream requests. Add the command line argument `--pass-authorization-header=false` or its equivalent in your oauth2-proxy config: https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview#command-line-options

**Comment by DrissiReda** [comment]
thank you very much, that works, but if I use the Auth header, can I restrict access to specific namespaces for different roles? Using kubernetes `rbac`

**Comment by AlanGreene** [comment]
Yes this should be possible by modifying the default roles / bindings provided by the Dashboard, although it's not officially supported / documented yet.

The `Authorization`, `Impersonate-User`, `Impersonate-Group`, and other related headers are passed through to the Kubernetes API server to handle auth.

The Dashboard will display errors (as you've seen) for anything it is unable to access or any action it's unable to perform. In future we'll handle this more gracefully by detecting the available permissions and only showing the navigation items and actions the user has access to. It's one of the major items on our roadmap and we're tracking it in a number of related issues, for example https://github.com/tektoncd/dashboard/issues/1420.
